### PR TITLE
[Improve][connector-jdbc] Add config item enable upsert by query

### PR DESCRIPTION
**Pull request by hailin0**
## Purpose of this pull request

Add `support_upsert_by_query_primary_key_exist` config item
Fix row field null value write to jdbc statement

## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [x] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/incubator-seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml)

### CODE REVIEW

**Review by liugddx — COMMENTED** [review]
This method has not been pressure tested?

**Comment by hailin0** [comment]
> This method has not been pressure tested?

The cdc event sink cannot be tested in e2e for the time being

File: seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/converter/AbstractJdbcRowConverter.java [hunk 1]
```diff
@@ -113,6 +113,12 @@ public PreparedStatement toExternal(SeaTunnelRowType rowType, SeaTunnelRow row,
         for (int fieldIndex = 0; fieldIndex < rowType.getTotalFields(); fieldIndex++) {
             SeaTunnelDataType<?> seaTunnelDataType = rowType.getFieldType(fieldIndex);
             int statementIndex = fieldIndex + 1;
+            Object fieldValue = row.getField(fieldIndex);
+            if (fieldValue == null) {
+                statement.setObject(statementIndex, null);
+                return statement;
```
Comment (liugddx) [inline seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/converter/AbstractJdbcRowConverter.java]: continue?

Comment (hailin0) [reply seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/converter/AbstractJdbcRowConverter.java]: thx, fixed

File: seatunnel-dist/src/main/assembly/assembly-bin-ci.xml [hunk 2]
```diff
@@ -184,7 +184,7 @@
                 <include>com.amazon.redshift:redshift-jdbc42:jar</include>
             </includes>
             <outputFileNameMapping>${artifact.file.name}</outputFileNameMapping>
-            <outputDirectory>/lib/jdbc_driver</outputDirectory>
+            <outputDirectory>/plugins/jdbc</outputDirectory>
```
Comment (hailin0) [inline seatunnel-dist/src/main/assembly/assembly-bin-ci.xml]: fix 
https://github.com/apache/incubator-seatunnel/pull/3689

**Review by EricJoy2048 — APPROVED** [review]

**Review by liugddx — APPROVED** [review]

**Pull request merged**